alf/algorithms/algorithm.py

```diff
@@ -1258,7 +1258,12 @@ def _make_time_major(nest):
 
         for u in range(num_updates):
             if mini_batch_size < batch_size:
-                indices = torch.randperm(batch_size)
+                # here we use numpy random.permutation to generate the permuted
+                # indices, as the cuda version of torch.randperm(n) seems to
+                # have a bug when n is a large number, generating negative or
+                # very large values that cause out of bound kernel error
+                # https://github.com/pytorch/vision/issues/3816
+                indices = np.random.permutation(batch_size)
```

Comment:
indices will be used several times at line 1267 and 1271. It's better to convert to tensor first `indices = torch.as_tensor(indices)`